### PR TITLE
8322279: Generational ZGC: Use ZFragmentationLimit and ZYoungCompactionLimit as percentage instead of multiples

### PR DESCRIPTION
**Pull request by lgxbslgx**
Hi all,

This patch fixes `ZHeuristics::significant_heap_overhead` and `ZHeuristics::significant_young_overhead` to use `ZFragmentationLimit` and `ZYoungCompactionLimit` as percentages instead of multiples.

Thanks for the review.

Best Regards,
-- Guoxiong

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8322279](https://bugs.openjdk.org/browse/JDK-8322279): Generational ZGC: Use ZFragmentationLimit and ZYoungCompactionLimit as percentage instead of multiples (**Bug** - P2)

### Reviewers
 * [Erik Österlund](https://openjdk.org/census#eosterlund) (@fisk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17139/head:pull/17139` \
`$ git checkout pull/17139`

Update a local copy of the PR: \
`$ git checkout pull/17139` \
`$ git pull https://git.openjdk.org/jdk.git pull/17139/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17139`

View PR using the GUI difftool: \
`$ git pr show -t 17139`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17139.diff">https://git.openjdk.org/jdk/pull/17139.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17139#issuecomment-1860210734)

### CODE REVIEW

**Comment by lgxbslgx** [comment]
Ping for review. This patch is related to https://github.com/openjdk/jdk/pull/17138. I would like to integrate these two patches at the same time.

**Review by fisk — APPROVED** [review]
Looks good.

**Comment by lgxbslgx** [comment]
Waiting for another review.

**Comment by fisk** [comment]
@lgxbslgx the change looks trivial; please feel free to integrate with only my review.

**Comment by lgxbslgx** [comment]
> @lgxbslgx the change looks trivial; please feel free to integrate with only my review.

Thanks for the review and guide. Integrating.

/integrate

**Pull request closed without merging**
